### PR TITLE
Fixes #4231: Adding support for svgz images 

### LINKED ISSUE #4231: Add support for svgz images
**Is your feature request related to a problem? Please describe.**
Oppia web is planning to add support for svgz images (which are just svg images with gzip compression) in order to reduce bandwidth and storage requirements for images.

**Describe the solution you'd like**
The Android app should be updated to support images ending with svgz by first decompressing them using built-in Android gzip utilities, and then passing them to the existing gzip pipeline (this may benefit from making use of a Glide transcoder).

**Describe alternatives you've considered**
The alternative would be to not support svgz at all, but that doesn't seem ideal since there are very clear benefits for both download and storage size (APK size for embedded lessons will barely change since APKs are already zip-compressed).

**Additional context**
None

### PR DESCRIPTION
**Pull request by supreme96**
## Explanation
Fixes #4231: Adding support for svgz images 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

SVGZ file used : https://raw.githubusercontent.com/Rioiv/svgz/16d5417b6bfcb18ea9e1d354e7dda3e73abad18f/1.svgz

### (manual test) current behavior of app loading a png file in the lesson.
original_behavior_loading_png_file

### (manual test) loading an svgz file without the change.

manual_test_failing_without_the_change

### (manual test) successfully loading the svgz file with the change.

svg_image

###  the new unit tests failing without the change.

unit_test_failing

### CODE REVIEW

**Comment by supreme96** [comment]
@BenHenning @seanlip PTAL.

File: utility/src/main/java/org/oppia/android/util/parser/image/UrlImageParser.kt
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/parser/image/UrlImageParser.kt]:
```suggestion
    val isSvg = machineLocale.run {
      imageUrl.endsWithIgnoreCase("svg") ||imageUrl.endsWithIgnoreCase("svgz")
    }
```

Can be simplified a bit.

File: utility/src/test/java/org/oppia/android/util/parser/image/UrlImageParserTest.kt
Comment (BenHenning) [inline utility/src/test/java/org/oppia/android/util/parser/image/UrlImageParserTest.kt]: For this & the other methods, perhaps instead it should be testLoadDrawable_svgz_... to differentiate with the other svg files?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @supreme96! I think this looks quite good! Just had a few comments--PtAL.

I'm also updating this to the latest develop and kicking off CI so that we can get CI results.

@seanlip are there any SVGZ images on the Oppia web side that we might be able to manually point to in order to test & verify that they can be loaded correctly in the pipeline (now that they're being supported)? If not, @supreme96 I suggest perhaps temporarily updating calls to loadDrawable() to force-load any sample SVGZ file from the internet, and then showing via a video that it loads correctly in the app (please include a link to the SVGZ file itself for reference). It'd be especially helpful to see a before & after video (that is, a video showing that the image does not load correctly before these fixes are in place).

Also, @supreme96 could you update the PR description to include screenshots of the new tests failing without the UrlImageParser changes in place? It's good to verify that the tests fail when expected so that we know they'll catch regressions for this logic in the future.

**Comment by seanlip** [comment]
> @seanlip are there any SVGZ images on the Oppia web side that we might be able to manually point to in order to test & verify that they can be loaded correctly in the pipeline (now that they're being supported)?

No, not yet, sorry. @supreme96, I would suggest following Ben's suggestion here. Thanks!

**Comment by supreme96** [comment]
> Thanks @supreme96! I think this looks quite good! Just had a few comments--PtAL.
> 
> I'm also updating this to the latest develop and kicking off CI so that we can get CI results.
> 
> @seanlip are there any SVGZ images on the Oppia web side that we might be able to manually point to in order to test & verify that they can be loaded correctly in the pipeline (now that they're being supported)? If not, @supreme96 I suggest perhaps temporarily updating calls to loadDrawable() to force-load any sample SVGZ file from the internet, and then showing via a video that it loads correctly in the app (please include a link to the SVGZ file itself for reference). It'd be especially helpful to see a before & after video (that is, a video showing that the image does not load correctly before these fixes are in place).
> 
> Also, @supreme96 could you update the PR description to include screenshots of the new tests failing without the UrlImageParser changes in place? It's good to verify that the tests fail when expected so that we know they'll catch regressions for this logic in the future.

@BenHenning Can you help me figure out where we are displaying svgs in the app? I can then test the change manually. I will also upload the requested screenshots and videos.

**Comment by supreme96** [comment]
@BenHenning Need some help. PTAL at the comment above.

**Comment by seanlip** [comment]
Any images in the "lesson" parts of the app are typically SVGs. E.g. the thumbnail images for graphics, the images in lessons, etc.

You'll need to look for these in the data supplied with the code, though. The above only explains how you can see an SVG image visually when you run Oppia Android locally.

**Comment by BenHenning** [comment]
Adding to @seanlip's comment, but you can look for oppia-noninteractive-image tags within the existing fractions or ratios lessons that are included in the app locally (see src/domain/main/assets) and either change one of those, or update one of the other lessons to include a tag pointing to svgz to show that it can load correctly.

I highly suggest verifying the code both with and without the fix in place (without the image shouldn't show up at all--this is a good verification to take a screenshot of to demonstrate that the lesson was, indeed, updated to use svgz).

**Comment by supreme96** [comment]
> Adding to @seanlip's comment, but you can look for oppia-noninteractive-image tags within the existing fractions or ratios lessons that are included in the app locally (see src/domain/main/assets) and either change one of those, or update one of the other lessons to include a tag pointing to svgz to show that it can load correctly.
> 
> I highly suggest verifying the code both with and without the fix in place (without the image shouldn't show up at all--this is a good verification to take a screenshot of to demonstrate that the lesson was, indeed, updated to use svgz).

Sure. I'll try that. Thanks :D

File: utility/src/main/java/org/oppia/android/util/parser/image/UrlImageParser.kt
Comment (supreme96) [reply utility/src/main/java/org/oppia/android/util/parser/image/UrlImageParser.kt]: Done.

File: utility/src/test/java/org/oppia/android/util/parser/image/UrlImageParserTest.kt
Comment (supreme96) [reply utility/src/test/java/org/oppia/android/util/parser/image/UrlImageParserTest.kt]: Done

**Comment by supreme96** [comment]
Hi @BenHenning PTAL, adding screenshots as requested. 

SVGZ file used : https://raw.githubusercontent.com/Rioiv/svgz/16d5417b6bfcb18ea9e1d354e7dda3e73abad18f/1.svgz

### (manual test) current behavior of app loading a png file in the lesson.
original_behavior_loading_png_file

### (manual test) loading an svgz file without the change.

manual_test_failing_without_the_change

### (manual test) successfully loading the svgz file with the change.

svg_image

###  the new unit tests failing without the change.

unit_test_failing

**Comment by supreme96** [comment]
> Thanks @supreme96! I think this looks quite good! Just had a few comments--PtAL.
> 
> I'm also updating this to the latest develop and kicking off CI so that we can get CI results.
> 
> @seanlip are there any SVGZ images on the Oppia web side that we might be able to manually point to in order to test & verify that they can be loaded correctly in the pipeline (now that they're being supported)? If not, @supreme96 I suggest perhaps temporarily updating calls to loadDrawable() to force-load any sample SVGZ file from the internet, and then showing via a video that it loads correctly in the app (please include a link to the SVGZ file itself for reference). It'd be especially helpful to see a before & after video (that is, a video showing that the image does not load correctly before these fixes are in place).
> 
> Also, @supreme96 could you update the PR description to include screenshots of the new tests failing without the UrlImageParser changes in place? It's good to verify that the tests fail when expected so that we know they'll catch regressions for this logic in the future.

Added.

**Review by BenHenning — APPROVED** [review]
Thanks @supreme96! I think this looks really good--well done on the screenshots demonstrating the behavior. My only nit is that the screenshots should go in the PR description (but I went ahead and did this to avoid needing another back-and-forth on the PR).

Just waiting on CI to pass, but otherwise the PR LGTM.

**Pull request merged**
